src/overlays/actors/ovl_En_Fall/z_en_fall.c

```diff
@@ -36,75 +64,820 @@ const ActorInit En_Fall_InitVars = {
+            break;
+        default:
+            this->scale = 0.16f;
+            break;
+    }
+
+    switch (EN_FALL_TYPE(&this->actor)) {
+        case EN_FALL_TYPE_LODMOON_NO_LERP:
+        case EN_FALL_TYPE_LODMOON:
+        case EN_FALL_TYPE_LODMOON_INVERTED_STONE_TOWER:
+            objectIndex = Object_GetIndex(&globalCtx->objectCtx, OBJECT_LODMOON);
+            break;
+        case EN_FALL_TYPE_MOONS_TEAR:
+            objectIndex = Object_GetIndex(&globalCtx->objectCtx, OBJECT_MOONSTON);
+            break;
+        case EN_FALL_TYPE_STOPPED_MOON_OPEN_MOUTH:
+            objectIndex = Object_GetIndex(&globalCtx->objectCtx, OBJECT_FALL2);
+            break;
+        default:
+            objectIndex = Object_GetIndex(&globalCtx->objectCtx, OBJECT_FALL);
+            break;
+    }
+
+    if (objectIndex < 0) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+    this->objIndex = objectIndex;
+    this->actionFunc = EnFall_Setup;
+}
+
+void EnFall_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+/**
+ * Finds the Termina Field moon so the Moon's Tear can spawn in the correct place.
+ */
+Actor* EnFall_MoonsTear_GetTerminaFieldMoon(GlobalContext* globalCtx) {
+    Actor* itemAction = globalCtx->actorCtx.actorList[ACTORCAT_ITEMACTION].first;
+
+    while (itemAction != NULL) {
+        if (itemAction->id == ACTOR_EN_FALL && EN_FALL_TYPE(itemAction) == EN_FALL_TYPE_TERMINA_FIELD_MOON) {
+            return itemAction;
+        }
+        itemAction = itemAction->next;
+    }
+    return NULL;
+}
+
+void EnFall_Setup(EnFall* this, GlobalContext* globalCtx) {
+    Actor* moon;
+
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->objIndex)) {
+        this->actor.objBankIndex = this->objIndex;
+        this->actionFunc = EnFall_Moon_PerformDefaultActions;
+        switch (EN_FALL_TYPE(&this->actor)) {
+            case EN_FALL_TYPE_TITLE_SCREEN_MOON:
+                this->actor.draw = EnFall_Moon_Draw;
+                this->actionFunc = EnFall_ClockTowerOrTitleScreenMoon_PerformCutsceneActions;
+                Actor_SetScale(&this->actor, this->scale);
+                break;
+            case EN_FALL_TYPE_STOPPED_MOON_CLOSED_MOUTH:
+                this->actor.draw = EnFall_Moon_Draw;
+                this->actionFunc = EnFall_StoppedClosedMouthMoon_PerformCutsceneActions;
+                Actor_SetScale(&this->actor, this->scale * 3.0f);
+                if ((gSaveContext.weekEventReg[0x19] & 2) == 0) {
+                    Actor_MarkForDeath(&this->actor);
+                }
+                break;
+            case EN_FALL_TYPE_CLOCK_TOWER_MOON:
+                this->actionFunc = EnFall_ClockTowerOrTitleScreenMoon_PerformCutsceneActions;
+                Actor_SetScale(&this->actor, this->scale * 3.0f);
+                this->actor.draw = EnFall_Moon_Draw;
+                if ((gSaveContext.weekEventReg[0x19] & 2) != 0) {
+                    Actor_MarkForDeath(&this->actor);
+                }
```

Comment:
```suggestion
                if (!(gSaveContext.weekEventReg[0x19] & 2)) {
                    Actor_MarkForDeath(&this->actor);
                }
                break;
            case EN_FALL_TYPE_CLOCK_TOWER_MOON:
                this->actionFunc = EnFall_ClockTowerOrTitleScreenMoon_PerformCutsceneActions;
                Actor_SetScale(&this->actor, this->scale * 3.0f);
                this->actor.draw = EnFall_Moon_Draw;
                if (gSaveContext.weekEventReg[0x19] & 2) {
                    Actor_MarkForDeath(&this->actor);
                }
```
This looks like a flag?